### PR TITLE
support hive view

### PR DESCRIPTION
**Pull request by cloud-fan**

### CODE REVIEW

**Comment by cloud-fan** [comment]
will add more comments later if we dicide to go this way.

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/CreateViewAsSelect.scala [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        throw new AnalysisException(s"$database.$viewName already exists.")
+      }
+    } else {
+      val tbl = if (newNames.nonEmpty) {
+        val sb = new StringBuilder
+        sb.append("SELECT ")
+        for (i <- 0 until newNames.length) {
+          if (i > 0) {
+            sb.append(", ")
+          }
```
Comment (marmbrus) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/CreateViewAsSelect.scala]: Nit: I'm not sure perf matters here enough to not just use {{mkString}} and concatenation.

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/CreateViewAsSelect.scala [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+          sb.append(HiveUtils.unparseIdentifier(tableDesc.schema(i).name))
+          sb.append(" AS ")
+          sb.append(HiveUtils.unparseIdentifier(newNames(i)))
+        }
+        sb.append(" FROM (")
+        sb.append(tableDesc.viewText.get)
```
Comment (marmbrus) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/CreateViewAsSelect.scala]: Could we use a similar trick and grab all the names from the analyzed plan `output` and wrap the raw query with a select to make us more robust to `*`s?

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveQl.scala [hunk 3]
```diff
@@ -563,6 +562,77 @@ https://cwiki.apache.org/confluence/display/Hive/Enhanced+Aggregation%2C+Cube%2C
...
+          "TOK_TABLEPROPERTIES",
+          "TOK_TABCOLNAME",
+          "TOK_VIEWPARTCOLS"),
+        children)
+
+      if (maybePartCols.isDefined) {
+        val sql = ctx.getTokenRewriteStream
+          .toString(query.parent.getTokenStartIndex, query.parent.getTokenStopIndex)
+        println(sql)
+        HiveNativeCommand(sql)
```
Comment (yhuai) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveQl.scala]: I guess you can use `NativePlaceholder`.

**Pull request closed without merging**